src/preferences/constants.h

```diff
@@ -1,20 +1,44 @@
 #pragma once
 
+// required for Qt-Macros
+#include <QObject>
+
 namespace mixxx {
 
+namespace preferences {
+
+namespace constants {
+Q_NAMESPACE
+
+// In order for this Q_NAMESPACE to work, all members of the namespace must
+// be declared here. see QTBUG-68611
+
```

Comment:
I think for the other devs, this would be most controversial aspect. Putting a QObject header into this header just for the enums is not great in terms of compile-time overhead, but I think its a necessary evil anyways in order to make these enums usable from within QML.

Reply:
Q_NAMESPACE is defined in qobjectdefs.h

Reply:
Tried that, but I got MOC errors.

Reply:
Tried 5.15.12. Though I may be misremembering what header I tried.